### PR TITLE
Edit and add tests for profile submodule #1711

### PR DESCRIPTION
**Pull request by willingc**
Add 3 tests for ``url2printably_short`` function.
Clean up formatting of ``profile/models.py``

Addresses #1711

### CODE REVIEW

File: mysite/profile/models.py [hunk 1]
```diff
@@ -98,46 +93,29 @@ class Person(models.Model):
...
-                                        generate_person_photo_path(
-                                            a, b, suffix="-thumbnail"),
-                                        default='',
-                                        null=True)
-
-    photo_thumbnail_30px_wide = models.ImageField(upload_to=
-                                                  lambda a, b: 'static/photos/profile-photos/' +
-                                                  generate_person_photo_path(
-                                                      a, b, suffix="-thumbnail-30px-wide"),
+    photo = models.ImageField(upload_to=lambda a, b: 'static/photos/profile-photos/' + generate_person_photo_path(a, b), default='')
```
Comment (ehashman) [inline mysite/profile/models.py]: Doesn't this violate PEP8 now? Seems too long.

File: mysite/profile/models.py [hunk 2]
```diff
@@ -277,18 +249,15 @@ def get_maintainer_portfolio_entries(self):
...
         # if you change this method, be sure to increment the version number in
         # the cache key above
-        return list(self.get_published_portfolio_entries().values_list(
-            'project__name', flat=True).distinct())
+        return list(self.get_published_portfolio_entries().values_list('project__name', flat=True).distinct())
 
     def get_display_names_of_nonarchived_projects(self):
-        return list(self.get_nonarchived_published_portfolio_entries().values_list(
-            'project__display_name', flat=True).distinct())
+        return list(self.get_nonarchived_published_portfolio_entries().values_list('project__display_name', flat=True).distinct())
```
Comment (ehashman) [inline mysite/profile/models.py]: Hm, again, this looks like it's going to make this not PEP8-compliant.

**Comment by ehashman** [comment]
@willingc would you mind making changes s.t. some of these changes are reverted to ensure PEP8 compliance? I haven't read through the whole thing but I don't want to keep commenting ">79 characters PEP8 violation" :P

**Comment by willingc** [comment]
@ehashman Yes they are long lines. I increased the line length while we refactor and move toward a more current Django. https://www.youtube.com/watch?v=wf-BqAjZb8M inspired

**Comment by ehashman** [comment]
This talk is awesome but nonetheless it is hard to read both content changes and formatting changes in the same review. I will try to do this piece by piece.

**Comment by willingc** [comment]
Hey @ehashman. Do you want me to break up the PR?

**Comment by ehashman** [comment]
That would be super helpful, yes!

**Comment by willingc** [comment]
@ehashman I broke this PR up into 2 PRs. This PR covers only changes to profile.tests.

Three significant change areas:
1. Imports have been reordered and double checked that they are in fact used.
2. Formatting changes - Line length relaxed to 99 characters to accommodate the long test names/filters/etc.
3. New tests added at the bottom of the file to increase coverage of the some uncovered models.

I hope this is easier to review :D

File: mysite/profile/tests.py [hunk 3]
```diff
@@ -235,18 +240,26 @@ class MockFetchPersonDataFromOhloh(object):
...
-     u'contributor_id': u'110891760646528'}
-], WebResponse())
+stumps_ohloh_results.return_value = (
+    [
+        {u'contributor_name': u'stump',
+         u'analysis_id': u'1145788',
+         u'man_months': u'11',
+         u'primary_language_nice_name': u'C',
+         u'contributor_id': u'2008814186590608'},
+        {u'contributor_name': u'John Stumpo',
```
Comment (ehashman) [inline mysite/profile/tests.py]: This { looks almost like it ran astray, might be better if they had their own lines.

File: mysite/profile/tests.py [hunk 4]
```diff
@@ -606,18 +606,18 @@ def test_delete_portfolio_entry(self):
         }
 
         self.assertEqual(response_decoded, expected_output)
-        self.assert_(PortfolioEntry.objects.get(pk=portfolio_entry.pk)
-                     .is_deleted)
+        self.assertTrue(PortfolioEntry.objects.get(pk=portfolio_entry.pk).is_deleted)
 
-    def test_delete_portfolio_entry_fails_when_portfolio_entry_doesnt_exist(
-            self):
```
Comment (ehashman) [inline mysite/profile/tests.py]: Wow this is silly. +1 to fix

File: mysite/profile/tests.py [hunk 5]
```diff
@@ -606,18 +606,18 @@ def test_delete_portfolio_entry(self):
...
+    def test_delete_portfolio_entry_fails_when_portfolio_entry_doesnt_exist(self):
         failing_pk = 0
         self.assertEqual(
-            PortfolioEntry.objects.filter(pk=failing_pk).count(), 0)
+            PortfolioEntry.objects.filter(pk=failing_pk).count(),
+            0
+        )
```
Comment (ehashman) [inline mysite/profile/tests.py]: This could probably occupy a single line (up with self.assertEqual)

File: mysite/profile/tests.py [hunk 6]
```diff
@@ -696,13 +697,13 @@ def test_add_citation_manually_with_bad_portfolio_entry(self):
 
         # Check that no citation was created.
         self.assertEqual(
-            Citation.untrashed.filter(url=input_data['url']).count(), 0,
+            Citation.untrashed.filter(url=input_data['url']).count(),
+            0,
             "Expected no citation to be created when you try "
             "to add one for someone else.")
```
Comment (ehashman) [inline mysite/profile/tests.py]: Maybe also give this ) its own line.

File: mysite/profile/tests.py [hunk 7]
```diff
@@ -716,10 +717,9 @@ def setUp(self):
 
         self.portfolio_entry = PortfolioEntry.objects.get_or_create(
             project=self.project,
-            person=Person.objects.get(user__username=self.user))[0]
-        citation = Citation(
-            portfolio_entry=self.portfolio_entry
-        )
+            person=Person.objects.get(user__username=self.user)
+        )[0]
+        citation = Citation(portfolio_entry=self.portfolio_entry)
```
Comment (ehashman) [inline mysite/profile/tests.py]: +1

File: mysite/profile/tests.py [hunk 8]
```diff
@@ -2466,20 +2487,68 @@ def test_actual_deletion_as_paulproteus(self):
...
+        self.assertTrue(mysite.profile.models.url2printably_short(url, cutoff))
+
+    def test_url2printably_short_equal_cutoff(self):
+        cutoff = 50
+        url = 'https://example.com/blogit/peanutbutter/jelly.html'
+        self.assertEqual(len(url), cutoff)
+        self.assertTrue(mysite.profile.models.url2printably_short(url, cutoff))
+
+    def test_url2printably_short_greater_cutoff(self):
+        cutoff = 50
```
Comment (ehashman) [inline mysite/profile/tests.py]: I guess you could declare cutoff as a static property of the class here so you don't have to declare it 3 times

**Comment by ehashman** [comment]
Looks like I can't merge this one, but it's been reviewed.

**Pull request closed without merging**